tests/foreman/cli/test_host.py

```diff
@@ -79,3 +82,205 @@ def test_create_libvirt_without_mac(self):
             u'puppet-proxy-id': self.puppet_proxy['id'],
             u'root-pass': host.root_pass,
         })
+
+
+class HostParameterTests(CLITestCase):
+    """Tests targeting host parameters"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create host to tests parameters for"""
+        super(HostParameterTests, cls).setUpClass()
+        cls.puppet_proxy = Proxy.list()[0]
+        cls.host = entities.Host()
+        cls.host.create_missing()
+        cls.host = Host.create({
+            u'architecture-id': cls.host.architecture.id,
+            u'domain-id': cls.host.domain.id,
+            u'environment-id': cls.host.environment.id,
+            u'location-id': cls.host.location.id,  # pylint:disable=no-member
+            u'mac': cls.host.mac,
+            u'medium-id': cls.host.medium.id,
+            u'name': cls.host.name,
+            u'operatingsystem-id': cls.host.operatingsystem.id,
+            # pylint:disable=E1101
+            u'organization-id': cls.host.organization.id,
+            u'partition-table-id': cls.host.ptable.id,
+            u'puppet-proxy-id': cls.puppet_proxy['id'],
+            u'root-pass': cls.host.root_pass,
+        })
+
+    def test_add_parameter_diff_names(self):
+        """@Test: Add host parameter with different valid names.
+
+        @Feature: Hosts
+
+        @Assert: Host parameter was successfully added with correct name.
+
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                name = name.lower()
+                Host.set_parameter({
+                    'host-id': self.host['id'],
+                    'name': name,
+                    'value': gen_string('alphanumeric'),
+                })
+                self.host = Host.info({'id': self.host['id']})
+                self.assertIn(name, self.host['parameters'].keys())
+
+    def test_add_parameter_diff_values(self):
```

Comment:
but right now it's more consistent with previous test - previous checked different names, this one checks different values.